src/components/AutoUpdateTime.js

```diff
@@ -22,85 +27,70 @@ const propTypes = {
-        this.state = {
-            currentUserLocalTime: this.getCurrentUserLocalTime(),
-        };
-    }
-
-    componentDidMount() {
-        this.updateCurrentTime();
-    }
-
-    componentDidUpdate() {
-        // Make sure the interval is up to date every time the component updates
-        this.updateCurrentTime();
-    }
-
-    componentWillUnmount() {
-        clearTimeout(this.timer);
-    }
-
+function AutoUpdateTime(props) {
     /**
      * @returns {moment} Returns the locale moment object
      */
-    getCurrentUserLocalTime() {
-        return DateUtils.getLocalMomentFromDatetime(
-            this.props.preferredLocale,
+    const getCurrentUserLocalTime = useCallback(() => (
+        DateUtils.getLocalMomentFromDatetime(
+            props.preferredLocale,
             null,
-            this.props.timezone.selected,
-        );
-    }
+            props.timezone.selected,
+        )
+    ), [props.preferredLocale, props.timezone.selected]);
+
+    const [currentUserLocalTime, setCurrentUserLocalTime] = useState(getCurrentUserLocalTime());
+    const timerRef = useRef(null);
 
     /**
      * @returns {string} Returns the timezone name in string, e.g.: GMT +07
      */
-    getTimezoneName() {
+    const getTimezoneName = useCallback(() => {
         // With non-GMT timezone, moment.zoneAbbr() will return the name of that timezone, so we can use it directly.
-        if (Number.isNaN(Number(this.state.currentUserLocalTime.zoneAbbr()))) {
-            return this.state.currentUserLocalTime.zoneAbbr();
+        if (Number.isNaN(Number(currentUserLocalTime.zoneAbbr()))) {
+            return currentUserLocalTime.zoneAbbr();
         }
 
         // With GMT timezone, moment.zoneAbbr() will return a number, so we need to display it as GMT {abbreviations} format, e.g.: GMT +07
-        return `GMT ${this.state.currentUserLocalTime.zoneAbbr()}`;
-    }
+        return `GMT ${currentUserLocalTime.zoneAbbr()}`;
+    }, [currentUserLocalTime]);
 
     /**
      * Update the user's local time at the top of every minute
      */
-    updateCurrentTime() {
-        if (this.timer) {
-            clearTimeout(this.timer);
-            this.timer = null;
+    const updateCurrentTime = useCallback(() => {
+        if (timerRef.current) {
+            clearTimeout(timerRef.current);
+            timerRef.current = null;
         }
-        const millisecondsUntilNextMinute = (60 - this.state.currentUserLocalTime.seconds()) * 1000;
-        this.timer = setTimeout(() => {
-            this.setState({
-                currentUserLocalTime: this.getCurrentUserLocalTime(),
-            });
+        const millisecondsUntilNextMinute = (60 - currentUserLocalTime.seconds()) * 1000;
+        timerRef.current = setTimeout(() => {
+            setCurrentUserLocalTime(getCurrentUserLocalTime());
         }, millisecondsUntilNextMinute);
-    }
+    }, [currentUserLocalTime, getCurrentUserLocalTime]);
```

Comment:
Hmmm... this method depends on `currentUserLocalTime` that is changed by this method, so every time the `setTimeout` callback is called, this method is re-created, and then called again in the `useEffect` below. Sounds like loop... and not the best design in terms of performance. You should avoid depending on `currentUsetLocalTime` state here.

Reply:
Responded in the other thread, but I do think we want an infinite loop here.